test/integration/scheduler/taint_test.go

```diff
@@ -689,13 +690,14 @@ func TestTaintBasedEvictions(t *testing.T) {
 				0.55,             // Unhealthy zone threshold
 				true,             // Run taint manager
 				true,             // Use taint based evictions
-				false,            // Enabled TaintNodeByCondition feature
```

Comment:
@ahg-g TestTaintBasedEvictions needs to disable the TaintNodeByCondition here

Reply:
We shouldn't be testing code paths that can only be enabled in tests. The test should be re-done to create the desired test state without needing to manually disable parts of the code.

@Huang-Wei I think you did work on this feature (TaintBasedEvictions), can you please take a look and advise how we can fix this? How should the test be re-written to test TaintBasedEvictions while TaintNodeByCondition is enabled.

Reply:
A little background on this:

I used to write an e2e test for TaintBasedEviction, which works well, but only on GCE/GKE.

So @ravisantoshgudimetla and @damemi proposed to convert that e2e test to an integration one, which can be portable on all providers. The integration is the current version of taint_test.go#TestTaintBasedEvictions. I reviewed and approved that PR #81856, but overlooked it set the TaintNodesByCondition flag to false.

I totally agree the test shouldn't depend on the false flag, I will take a close look to make the integration test work (without setting the flag to false) first.

Reply:
can we revert the test in this PR, and @Huang-Wei work in parallel to add it back without the flag so that we don't block this PR? it was not practically useful anyway since it didn't test the actual setup we have right now for the feature flags.

Reply:
@ahg-g I just proposed a PR #84036 solving that issue. After it's merged, @draveness should be able to proceed on this one.